### PR TITLE
[Frontend] Add `reasoning_effort` and `parallel_tool_calls` to `extra_args` of `SamplingParams`.

### PR DESCRIPTION
**Pull request by sanghoon-yn**
## Purpose

~~I suggest adding two fields, `reasoning_effort` and `parallel_tool_calls`, to `SamplingParams` to enable logits processors to support the OpenAI-compatible parameters by controlling token generation.~~

I suggest adding two fields, `reasoning_effort` and `parallel_tool_calls`, to `extra_args` of `SamplingParams` to enable logits processors to support the OpenAI-compatible parameters by controlling token generation.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

---

> [!NOTE]
> Adds OpenAI-compatible routing of request fields into sampling.
> 
> - Passes `reasoning.effort` and `parallel_tool_calls` into `SamplingParams.extra_args` for both `ResponsesRequest.to_sampling_params` and `ChatCompletionRequest.to_sampling_params`
> - Expands accepted `reasoning_effort` values to include `"minimal"` (in `harmony_utils.get_system_message` and `ChatCompletionRequest`)
> - Preserves existing behavior; no other sampling fields or APIs changed
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7e1d192dbd58df3215bec455e87a296f3c3837a7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Enables OpenAI-compatible control fields to reach sampling and expands accepted reasoning settings.
> 
> - Passes `reasoning.effort` and `parallel_tool_calls` into `SamplingParams.extra_args` in both `ResponsesRequest.to_sampling_params` and `ChatCompletionRequest.to_sampling_params`
> - Expands `reasoning_effort` to include `"minimal"` in `ChatCompletionRequest` and `harmony_utils.get_system_message`
> - No other sampling parameters or APIs changed
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit eecb31ff77e33e86a09954132b226d75fd62eda1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by DarkLight1337 — COMMENTED** [review]
I think this should be part of `extra_args` since none of the in-tree logits processors use these arguments. cc @Isotr0py since you did similar thing for `NGramPerReqLogitsProcessor`

File: vllm/entrypoints/openai/protocol.py [hunk 1]
```diff
@@ -816,6 +823,10 @@ def to_sampling_params(
         if self.kv_transfer_params:
             # Pass in kv_transfer_params via extra_args
             extra_args["kv_transfer_params"] = self.kv_transfer_params
+        if self.reasoning_effort is not None:
+            extra_args["reasoning_effort"] = self.reasoning_effort
+        extra_args["parallel_tool_calls"] = self.parallel_tool_calls
```
Comment (DarkLight1337) [inline vllm/entrypoints/openai/protocol.py]:
```suggestion
        if self.parallel_tool_calls is not None:
            extra_args["parallel_tool_calls"] = self.parallel_tool_calls
```

File: vllm/entrypoints/openai/protocol.py [hunk 2]
```diff
@@ -432,6 +432,12 @@ def to_sampling_params(
             elif response_format.type == "json_object":
                 raise NotImplementedError("json_object is not supported")
 
+        extra_args = dict(
```
Comment (DarkLight1337) [inline vllm/entrypoints/openai/protocol.py]: Same

Comment (sanghoon-yn) [reply vllm/entrypoints/openai/protocol.py]: fixed at https://github.com/vllm-project/vllm/pull/31952/commits/7e1d192dbd58df3215bec455e87a296f3c3837a7

File: vllm/entrypoints/openai/protocol.py (hunk 1 above)
Comment (sanghoon-yn) [reply vllm/entrypoints/openai/protocol.py]: Thank you for the review.
It's fixed at https://github.com/vllm-project/vllm/pull/31952/commits/7e1d192dbd58df3215bec455e87a296f3c3837a7 .
